### PR TITLE
License Update

### PR DESCRIPTION
**Pull request by JaxkDev**
Supersedes #319 

Changes a few things listed below:
- Featured group removed.
- Deprecated list added for old spdx ID's
- View License temporarily disabled, also needs updating to new endpoint.
- License link on plugin page (clicking license ID) redirects to spdx.org endpoint.
- All OSI approved licenses from https://spdx.org/licenses/licenses.json are now listed. (fixing #236)

### CODE REVIEW

**Comment by JaxkDev** [comment]
Slapper License Detection working: Screenshot 2023-06-02 at 19 59 47

Just a few of the options: Screenshot 2023-06-02 at 20 00 45

File: src/poggit/module/LicensesAjax.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+use poggit\utils\internet\Curl;
+use function json_encode;
+use const JSON_UNESCAPED_SLASHES;
+use const JSON_UNESCAPED_UNICODE;
+
+
+/*
+ * This is a copy pasta ajax, since I can't find any user calls from js file.
+ * If its OK to do calls from client js, then this can be removed and replaced client side.
```
Comment (SOF3) [inline src/poggit/module/LicensesAjax.php:31]: Not sure if there are any weird CORS requirements, let's just proxy it here.

File: src/poggit/release/submit/PluginSubmission.php [hunk 2]
```diff
@@ -321,8 +332,38 @@ public function processArtifact() {
         $py["version"] = $this->version;
         $py["api"] = SubmitFormAjax::rangesToApis($this->spoons);
         file_put_contents($pharUrl . "plugin.yml", yaml_emit($py));
-        if(!is_file($pharUrl . "LICENSE")) {
-            // TODO insert license here
+        if((!is_file($pharUrl . "LICENSE") || !is_file($pharUrl . "LICENSE.md") || !is_file($pharUrl . "LICENSE.MD")) && $this->license->type !== null && $this->license->type !== "custom" && $this->license->type !== "none") {
```
Comment (SOF3) [inline src/poggit/release/submit/PluginSubmission.php]: Maybe `LICENSE.txt` too?

**Review by SOF3 — APPROVED** [review]

**Comment by JaxkDev** [comment]
Ready, @SOF3 final review / pull etc

File: src/poggit/release/submit/PluginSubmission.php [hunk 3]
```diff
@@ -332,7 +332,7 @@ public function processArtifact() {
         $py["version"] = $this->version;
         $py["api"] = SubmitFormAjax::rangesToApis($this->spoons);
         file_put_contents($pharUrl . "plugin.yml", yaml_emit($py));
-        if((!is_file($pharUrl . "LICENSE") || !is_file($pharUrl . "LICENSE.md") || !is_file($pharUrl . "LICENSE.MD")) && $this->license->type !== null && $this->license->type !== "custom" && $this->license->type !== "none") {
+        if((!is_file($pharUrl . "LICENSE") || !is_file($pharUrl . "LICENSE.md") || !is_file($pharUrl . "LICENSE.MD") || !is_file($pharUrl . "LICENSE.txt")) && $this->license->type !== null && $this->license->type !== "custom" && $this->license->type !== "none") {
```
Comment (dktapps) [inline src/poggit/release/submit/PluginSubmission.php]: does github recognize the file with a lowercase name?

Comment (JaxkDev) [reply src/poggit/release/submit/PluginSubmission.php]: This parts GitHub irrelevant, but I don't know, I think so.

Comment (dktapps) [reply src/poggit/release/submit/PluginSubmission.php]: I think we should probably accept lowercase filenames.

Comment (JaxkDev) [reply src/poggit/release/submit/PluginSubmission.php]: [58375f6](https://github.com/poggit/poggit/pull/325/commits/58375f61b23ec4af46e85abb0e5fbce472601a46)

File: src/poggit/release/submit/PluginSubmission.php [hunk 4]
```diff
@@ -332,7 +332,17 @@ public function processArtifact() {
         $py["version"] = $this->version;
         $py["api"] = SubmitFormAjax::rangesToApis($this->spoons);
         file_put_contents($pharUrl . "plugin.yml", yaml_emit($py));
-        if((!is_file($pharUrl . "LICENSE") || !is_file($pharUrl . "LICENSE.md") || !is_file($pharUrl . "LICENSE.MD") || !is_file($pharUrl . "LICENSE.txt")) && $this->license->type !== null && $this->license->type !== "custom" && $this->license->type !== "none") {
+
+        $licenses = ["LICENSE", "LICENSE.md", "LICENSE.MD", "LICENSE.txt", "LICENSE.TXT", "license", "license.md", "license.MD", "license.txt", "license.TXT"];
```
Comment (dktapps) [inline src/poggit/release/submit/PluginSubmission.php:336]: ever heard of strtolower?

Comment (JaxkDev) [reply src/poggit/release/submit/PluginSubmission.php:336]: the file extensions though, I have lol

**Review by SOF3 — APPROVED** [review]

**Comment by SOF3** [comment]
Merged, but not pulling yet since I might have to check PHP version compatibility first.

**Comment by JaxkDev** [comment]
> Merged, but not pulling yet since I might have to check PHP version compatibility first.

Hm?
I’ve been running it locally on same PHP 8.0 as prod

**Pull request merged**